app/src/main/java/com/rajatgoyal/puzzle15/model/GamePlay.java

```diff
@@ -0,0 +1,34 @@
+package com.rajatgoyal.puzzle15.model;
+
+/**
+ * @author rajatgoyal715
+ */
+public class GamePlay {
+    private int score;
+    private int moves;
+    private Time time;
+
+    public GamePlay(int moves, Time time) {
+        this.moves = moves;
+        this.time = time;
+        this.score = (int)((double)100 * Math.exp(- (double) (moves + time.toSeconds()) / (double) 100));
+    }
+
+    public GamePlay(int score, int moves, Time time) {
+        this.score = score;
+        this.moves = moves;
+        this.time = time;
+    }
+
+    public int getScore() {
```

Comment:
Have a look at https://projectlombok.org/ It's a helpful tool for especially Model classes. It has annotations to create Getters and Setters and more methods like that

Reply:
Thanks for the suggestion. We can totally use Project Lombok. But if we move existing code to Kotlin, is there any need to use Project Lombok then?

Reply:
Nope

Reply:
Use combination of Kotlin and Java for now, conversio of existing code could potentially cause some breakages